front/components/assistant/conversation/input_bar/InputBar.tsx

```diff
@@ -82,19 +76,40 @@ export function AssistantInputBar({
 
   const [isAnimating, setIsAnimating] = useState<boolean>(false);
   const { animate, selectedAssistant } = useContext(InputBarContext);
+  const animationTimeoutRef = useRef<NodeJS.Timeout | null>(null);
+
   useEffect(() => {
-    let timeoutId: NodeJS.Timeout;
     if (animate && !isAnimating) {
       setIsAnimating(true);
-      timeoutId = setTimeout(() => setIsAnimating(false), 1500);
+
+      // Clear any existing timeout to ensure animations do not overlap.
+      if (animationTimeoutRef.current) {
+        clearTimeout(animationTimeoutRef.current);
+      }
+
+      // Set timeout to set setIsAnimating to false after the duration.
+      animationTimeoutRef.current = setTimeout(() => {
+        setIsAnimating(false);
+        // Reset the ref after the timeout clears.
+        animationTimeoutRef.current = null;
+      }, 700);
     }
+  }, [animate, isAnimating]);
 
-    // Cleanup function to clear the timeout
+  // Cleanup timeout on component unmount.
+  useEffect(() => {
     return () => {
-      if (timeoutId) {
-        clearTimeout(timeoutId);
+      if (animationTimeoutRef.current) {
+        clearTimeout(animationTimeoutRef.current);
       }
     };
+  }, []);
+
+  useEffect(() => {
```

Comment:
Looks like this code is duplicated with line 85, no?

Reply:
oopsy